ONSdigital/census-rm-docker-dev
# Use JAVA_TOOL_OPTIONS for env var

##### DO NOT MERGE YET: All PRs for branch `use-least-privilege` in each of the Java repos need to be merged first.

# Motivation and Context

The Kubernetes deployment config was [recently updated](https://github.com/ONSdigital/census-rm-kubernetes/pull/14) to use `JAVA_TOOL_OPTIONS` in favour of `JAVA_OPTS`. The Dockerfile for each affected repo was then updated to remove `JAVA_OPTS`, e.g. [here](https://github.com/ONSdigital/census-rm-case-service/pull/3/files)

As a result, we were unable to attach a remote debug session in our IDEs when using the docker-compose from this repo.

# What has changed



Quick change to `docker-compose.yml` to use `JAVA_TOOL_OPTIONS` instead

# How to test?



* Checkout [this branch](https://github.com/ONSdigital/census-rm-case-service/tree/use-least-privilege) (or the same branch on any other related RM Java app).
* Create a Docker image for that branch: `mvn clean install -Ddocker.skip -DskipITs -Dmaven.test.skip=true`
* Update the `docker-compose.yml` file to use the image built above.
* `make up`
* Open your IDE and ensure that you're able to attach a debug session when running.
